### PR TITLE
WIP: ENH: Wrap CastImageFilter for VectorImage <-> Vector

### PR DESCRIPTION
**Pull request by thewtex**
Addresses #1133

### CODE REVIEW

File: Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx [hunk 1]
```diff
@@ -152,7 +153,7 @@ CastImageFilter<TInputImage, TOutputImage>::DynamicThreadedGenerateDataDispatche
     {
       const InputPixelType & inputPixel = inputIt.Get();
       OutputPixelType        value;
```
Comment (blowekamp) [inline Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx]: This will not get set to the correct length for vector pixels.

Comment (blowekamp) [reply Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx]: Outside the loops add something like:

```
OutputPixelType value;
NumericTraits<OutputPixelType>::SetLength(value, componentsPerPixels)
```

I'm not sure if its a `NumericTraits` or some other pixel traits class needed to set the  number components for all types OK.

Pulling it out size the loop also remove per-pixel allocation which is a performance killer.

**Comment by thewtex** [comment]
Superceded by https://github.com/InsightSoftwareConsortium/ITK/pull/2073/files

File: Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx (hunk 1 above)
Comment (thewtex) [reply Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx:155]: @blowekamp thanks for the note.

@Leengit it looks this addition was missed when with https://github.com/InsightSoftwareConsortium/ITK/pull/2073 -- could you please add it in a follow-up patch?

**Pull request closed without merging**